### PR TITLE
Issue/#172 unit tests

### PR DESCRIPTION
**Pull request by dem92**

### CODE REVIEW

File: nest/src/issue/issue.controller.spec.ts [hunk 1]
```diff
@@ -1,20 +1,48 @@
...
+/**
+ * The issue controller contains next to no logic, so there is little point to testing all of
+ * the functions. This file is only here to show how it could be done.
+ */
+describe('IssueController', () => {
+    let controller: IssueController;
+    let service: IssueService;
+
+    beforeEach(async () => {
+        // Nest provides utilities for createing a testing module to test classes https://docs.nestjs.com/fundamentals/testing#testing-utilities
```
Comment (MiniMarker) [inline nest/src/issue/issue.controller.spec.ts:17]: creating*

File: nest/src/issue/issue.service.spec.ts [hunk 2]
```diff
@@ -1,18 +1,142 @@
...
+import { issueRepositoryMock } from './mocks/issue.repository.mock';
+
+describe('IssueService', () => {
+    let service: IssueService;
+    let repository: IssueRepository;
+    let mockIssue: Issue;
+    let readSpy: jest.SpyInstance;
+
+    beforeEach(async () => {
+        // Nest provides utilities for createing a testing module to test classes https://docs.nestjs.com/fundamentals/testing#testing-utilities
```
Comment (MiniMarker) [inline nest/src/issue/issue.service.spec.ts:17]: creating*

File: nest/src/issue/issue.controller.spec.ts (hunk 1 above)
Comment (dem92) [reply nest/src/issue/issue.controller.spec.ts]: Fixed!

File: nest/src/issue/issue.service.spec.ts (hunk 2 above)
Comment (dem92) [reply nest/src/issue/issue.service.spec.ts]: Fixed!

**Pull request merged**